packages/safe-tools-client/app/services/scheduled-payment-sdk.ts

```diff
@@ -106,38 +106,49 @@ export default class SchedulePaymentSDKService extends Service {
     gasToken: TokenDetail
   ): Promise<ServiceGasEstimationResult> {
     const scheduledPaymentModule = await this.getSchedulePaymentModule();
-
-    const gasEstimationResult = await scheduledPaymentModule.estimateGas(
-      scenario,
-      {
-        safeAddress: this.safes.currentSafe?.address,
-        tokenAddress: token.address,
-        gasTokenAddress: gasToken.address,
-        hubUrl: config.hubUrl,
+    const retryAttempt = 0;
```

Comment:
I don't see where retryAttempt is getting incremented.

Consider extracting a tested method `withRetry(maxAttempts, func)` or adopting a library like https://github.com/normartin/ts-retry-promise

Reply:
updated